src/diffusers/schedulers/scheduling_ddim_flax.py

```diff
@@ -322,5 +322,22 @@ def add_noise(
         noisy_samples = sqrt_alpha_prod * original_samples + sqrt_one_minus_alpha_prod * noise
         return noisy_samples
 
+    def get_velocity(
+        self, 
+        sample: jnp.ndarray, 
+        noise: jnp.ndarray, 
+        timesteps: jnp.ndarray
+    ) -> jnp.ndarray:
+        sqrt_alpha_prod = self.alphas_cumprod[timesteps] ** 0.5
+        sqrt_alpha_prod = sqrt_alpha_prod.flatten()
+        sqrt_alpha_prod = broadcast_to_shape_from_left(sqrt_alpha_prod, sample.shape)
+
+        sqrt_one_minus_alpha_prod = (1 - self.alphas_cumprod[timesteps]) ** 0.5
+        sqrt_one_minus_alpha_prod = sqrt_one_minus_alpha_prod.flatten()
+        sqrt_one_minus_alpha_prod = broadcast_to_shape_from_left(sqrt_one_minus_alpha_prod, sample.shape)
+
+        velocity = sqrt_alpha_prod * noise - sqrt_one_minus_alpha_prod * sample
+        return velocity
```

Comment:
Looks good, but we'll need to refactor this into a common function so like `add_noise_common` cf #1661

We should add a `get_velocity_common` function to `scheduling_flax_util.py` like `add_noise_common` https://github.com/huggingface/diffusers/blob/main/src/diffusers/schedulers/scheduling_utils_flax.py#L245 and then call that function from `get_velocity`.


after this, in the training script, we'll need to call this method using
```python
noise_scheduler.get_velocity(noise_scheduler_state, latents, noise, timesteps)
```

Reply:
I'd suggest merging PR [#1783](https://github.com/huggingface/diffusers/pull/1783) first, the changes there are clearly the correct ones.

Reply:
cool, this PR was opened first so I preferred this, but if you are fine with it, we'll go ahead with #1783

Reply:
I am fine with it, go ahead.